### PR TITLE
InitCommand - If the new `key` and `<file>` don't match, prompt for confirmation

### PR DESCRIPTION
**Pull request by totten**
Screen Shot 2022-02-12 at 3 58 44 AM

### CODE REVIEW

**Comment by colemanw** [comment]
Looks good :+1:

File: src/CRM/CivixBundle/Command/InitCommand.php [hunk 1]
```diff
@@ -32,11 +32,13 @@ protected function configure() {
...
         "\n" .
         "<comment>Identification:</comment>\n" .
-        "  Keys must be lowercase alphanumeric (with dashes and underscores allowed).\n" .
+        "  Keys should be lowercase alphanumeric with underscores. Additionally, dots and dashes\n" .
+        "  are also legal - but they are discouraged for new extensions.\n" .
         "\n" .
-        "  Optionally, you may use a Java-style prefix (reverse domain name).\n" .
+        "  Historically, the key often included a Java-style prefix (reverse domain name).\n" .
+        "  The benefit of the prefix was mostly cosmetic. It did not provide consistent\n" .
+        "  isolation between similarly named extensions. It is now recommended to skip\n" .
```
Comment (demeritcowboy) [inline src/CRM/CivixBundle/Command/InitCommand.php:40]: Is there some text missing after this? I get this from civix help generate:module

```
Help:
  Create a new CiviCRM Module-Extension (Regenerate module.civix.php if ext.name not specified)

  Identification:
    Keys should be lowercase alphanumeric with underscores. Additionally, dots and dashes
    are also legal - but they are discouraged for new extensions.

    Historically, the key often included a Java-style prefix (reverse domain name).
    The benefit of the prefix was mostly cosmetic. It did not provide consistent
    isolation between similarly named extensions. It is now recommended to skip


  Examples:
    civix generate:module foo-bar
    civix generate:module foo-bar --license=AGPL-3.0 --author="Alice" --email="alice@example.org"
    civix generate:module org.example.foo-bar
```

**Comment by totten** [comment]
Thank you @demeritcowboy !

I've done some more copy-edits on both the help screen and the interactive screen. Hopefully, all the sentences are complete... 🙃 

Screen Shot 2022-02-14 at 1 49 01 PM

Screen Shot 2022-02-14 at 1 48 40 PM

File: src/CRM/CivixBundle/Command/InitCommand.php [hunk 2]
```diff
@@ -103,6 +107,30 @@ protected function execute(InputInterface $input, OutputInterface $output) {
       $output->writeln('<error>Unrecognized license (' . $ctx['license'] . ')</error>');
       return;
     }
+
+    if ($ctx['fullName'] !== $ctx['mainFile']) {
+      $output->writeln("");
+      $output->writeln("<info>ALERT: The requested command requires split-naming.</info>");
```
Comment (demeritcowboy) [inline src/CRM/CivixBundle/Command/InitCommand.php:113]: Overall looks good. A little bit of a mismatch now where it says "ALERT" but has green info styling. But not a blocker.

File: src/CRM/CivixBundle/Command/InitCommand.php [hunk 3]
```diff
@@ -103,6 +107,30 @@ protected function execute(InputInterface $input, OutputInterface $output) {
...
+      $output->writeln("         <comment>Example Usage:</comment> addStyleFile('{$ctx['fullName']}', 'example.css')");
+      $output->writeln("   <comment>\"File\"</comment>: Appears in PHP files+functions. No Java-style prefix.");
+      $output->writeln("         <comment>Requested Value:</comment> {$ctx['mainFile']}");
+      $output->writeln("         <comment>Example Usage:</comment> function {$ctx['mainFile']}_civicrm_config() {}");
+      $output->writeln("");
+      $output->writeln("Many developers find it easier to use matching names, but the");
+      $output->writeln("requested command requires splitting the names. You may continue with");
+      $output->writeln("split names, or you may cancel and try again with a simpler name.");
+      $output->writeln("");
+      if (!$this->confirm($input, $output, "Continue with current (split) name? [Y/n] ")) {
```
Comment (demeritcowboy) [inline src/CRM/CivixBundle/Command/InitCommand.php:129]: This doesn't work on windows but that's come up elsewhere so isn't something new, just is new in this command. The workaround is just don't use dots.

File: src/CRM/CivixBundle/Command/InitCommand.php (hunk 2 above)
Comment (totten) [reply src/CRM/CivixBundle/Command/InitCommand.php]: Tweaked:

Screen Shot 2022-02-14 at 2 29 26 PM

Comment (demeritcowboy) [reply src/CRM/CivixBundle/Command/InitCommand.php]: Looks good 👍

File: src/CRM/CivixBundle/Command/InitCommand.php (hunk 3 above)
Comment (totten) [reply src/CRM/CivixBundle/Command/InitCommand.php:129]: Hmm, curious. I don't have a good setup to try it out. FWIW, that's  a [thin wrapper](https://github.com/totten/civix/blob/master/src/CRM/CivixBundle/Command/AbstractCommand.php#L18-L29) around [Symfony `ConfirmationQuestion`](https://symfony.com/doc/4.4/components/console/helpers/questionhelper.html). Theories:

* The command somehow gets option`--yes`.
* There's a discrepancy in the version of `symfony/console`. (Ex: My local has `v4.4.34`)
* There's an upstream issue with `ConfirmationQuestion` in the active version of  `symfony/console`.

Comment (demeritcowboy) [reply src/CRM/CivixBundle/Command/InitCommand.php:129]: Ahh - symfony/console version. I must have not run composer install anytime recently.

**Pull request merged**